sklearn/utils/estimator_checks.py

```diff
@@ -798,6 +818,14 @@ def _check_transformer(name, transformer_orig, X, y):
         else:
             X_pred2 = transformer.transform(X)
             X_pred3 = transformer.fit_transform(X, y=y_)
+        # raises error on malformed input for transform
+        if hasattr(X, 'T') and not _safe_tags(transformer, "stateless"):
+            # If it's not an array, it does not have a 'T' property
+            assert_raises(ValueError, transformer.transform, X.T)
```

Comment:
This is not the assertion that's failing. The one at the end of this function is the one that's failing on stateless transformers, though it is a repeat of this with more detail.